gtk4/src/subclass/frame.rs

```diff
@@ -0,0 +1,55 @@
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+use super::widget::WidgetImpl;
+use crate::{Allocation, Frame, Widget};
+
+pub trait FrameImpl: FrameImplExt + WidgetImpl {
+    fn compute_child_allocation(&self, frame: &Self::Type) -> Allocation {
+        self.parent_compute_child_allocation(frame)
+    }
+}
+
+pub trait FrameImplExt: ObjectSubclass {
+    fn parent_compute_child_allocation(&self, frame: &Self::Type) -> Allocation;
+}
+
+impl<T: FrameImpl> FrameImplExt for T {
+    fn parent_compute_child_allocation(&self, frame: &Self::Type) -> Allocation {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkFrameClass;
+            let f = (*parent_class)
+                .compute_child_allocation
+                .expect("No parent class impl for \"compute_child_allocation\"");
+            let mut allocation = Allocation::uninitialized();
```

Comment:
Not a fan of this and we want to get rid of `uninitialiezd()` in the long term.

```rust
let mut allocation = std::mem::MaybeUninit::uninit();
f(..., allocation.as_mut_ptr());
allocation.assume_init()
```

Reply:
Hmm that probably does not work here. The last would have to be `Allocation(allocation.assume_init())` or something like that.

Maybe something for another time...